### PR TITLE
linter: added handling of in_array and array_search functions

### PR DESCRIPTION
**Pull request by i582**
If the first argument is of type string, then the third argument is required for correctness.

### CODE REVIEW

**Comment by quasilyte** [comment]
I think there is a typo in the commit message: it should be `in_array`, not `is_array`.
GitHub doesn't allow to comment on commit messages (Gerrit does).

File: src/cmd/embeddedrules/rules.php [hunk 1]
```diff
@@ -238,3 +238,52 @@ function argsOrder() {
...
+        false == $_;
+        $_ == null;
+        null == $_;
+        $x == $y;
+    }
+
+    /**
+     * @warning non-strict comparison (use !==)
+     * @type string $x
+     * @type string $y
```
Comment (quasilyte) [inline src/cmd/embeddedrules/rules.php:266]: I'm not sure what these type filters are supposed to do.
Maybe we can remove them?
Patterns below don't have `$x` or `$y` matcher variables.
I would guess it's a copy/paste error.

File: src/cmd/embeddedrules/rules.php [hunk 2]
```diff
@@ -238,3 +238,52 @@ function argsOrder() {
...
+/**
+ * @comment Report not-strict-enough comparisons.
+ * @before  in_array("what", $s)
+ * @after   in_array("what", $s, true)
+ */
+function strictCmp() {
+    /**
+     * @warning non-strict comparison (use ===)
+     * @type string $x
+     * @type string $y
```
Comment (quasilyte) [inline src/cmd/embeddedrules/rules.php:251]: Same as below. Please remove them if you don't need them.

**Comment by quasilyte** [comment]
If conflicts are solved and this rule was tested on the vkcom, then it can be merged.

**Comment by i582** [comment]
Closed due #830

**Pull request closed without merging**